README.md

```diff
@@ -11,22 +11,35 @@
-Preferences are stored as TOML dictionaries and are, by default, stored within a `(Julia)LocalPreferences.toml` file next to the currently-active project.
-If a preference is "exported", it is instead stored within the `(Julia)Project.toml`.
-The intention is to allow shared projects to contain shared preferences, while allowing for users themselves to override those preferences with their own settings in the `LocalPreferences.toml` file, which should be `.gitignore`d as the name implies.
+
+Note that the package can be installed on Julia v1.0+ but is only functional on Julia v1.6+.
+
+## Project-specific vs package-wide preferences
+
+Preferences are stored as TOML dictionaries and are, by default, stored within a `(Julia)LocalPreferences.toml` file next to the currently-active project. This results in *project-specific*
+preferences, meaning that different projects making use of the same package can set up
+different, non-conflicting preferences.
 
 Preferences can be set with depot-wide defaults; if package `Foo` is installed within your global environment and it has preferences set, these preferences will apply as long as your global environment is part of your [`LOAD_PATH`](https://docs.julialang.org/en/v1/manual/code-loading/#Environment-stacks).
 Preferences in environments higher up in the environment stack get overridden by the more proximal entries in the load path, ending with the currently active project.
 This allows depot-wide preference defaults to exist, with active projects able to merge or even completely overwrite these inherited preferences.
 See the docstring for `set_preferences!()` for the full details of how to set preferences to allow or disallow merging.
 
+In contrast, *package-wide* preferences are stored within within the package's own `(Julia)Project.toml` file. Such preferences apply to all users of the package, regardless of the active project.
+
+You can control which kind of preference you create; this is discussed in the API subsections below.
+
+## Run-time vs compile-time preferences
+
 Preferences that are accessed during compilation are automatically marked as compile-time preferences, and any change recorded to these preferences will cause the Julia compiler to recompile any cached precompilation `.ji` files for that module.
 This allows preferences to be used to influence code generation.
 When your package sets a compile-time preference, it is usually best to suggest to the user that they should restart Julia, to allow recompilation to occur.
 
-Note that the package can be installed on Julia v1.0+ but is only functional on Julia v1.6+.
+If you call `load_preference` (or its macro variant `@load_preference`) from "top-level" in the package,
+this is a compile-time preference. Otherwise (e.g., if it is "buried" inside a function, and that
+function doesn't get executed at top-level), it is a run-time preference. See examples in the first API section below.
```

Comment:
This isn't exactly true; a preference can still be loaded from within a function and used at compile time if that function is invoked during precompilation time.

Reply:
"and that function doesn't get executed at top-level"? But obviously it's not clear.

Reply:
Hmm, yeah.  I'm honestly not sure it matters much to draw a distinction between compile-time and non-compile-time preferences; perhaps we should just have a note that says "if you use a preference at compile time, you will have to restart Julia before the changes will be seen" and leave it at that?

Reply:
Alternatively, perhaps we could instead have `set_preferences()` spit out a `@warn()` when it notices a compile-time preference has been changed?

Reply:
I like that idea!

We still might need to discuss this issue somewhere, otherwise people might be mystified about why it sometimes requires a restart and why it sometimes doesn't. (Might feel like a bug when it doesn't warn.)